app.rb

```diff
@@ -0,0 +1,153 @@
+    end
+  end
+
+  def list_people
+    @person.each do |p|
+      puts "[#{p.class.name}] Name: #{p.name}, ID: #{p.id}, Age: #{p.age}"
+    end
+  end
+
+  def create_book
+    print 'Title:'
+    title = gets.chomp
+    print 'author:'
+    author = gets.chomp
+    @books.push(Book.new(title, author))
+    puts 'Book Created Successfully'
+  end
+
+  def check_number(msg)
+    number = 0
+    loop do
+      print msg
+      input = gets.chomp.to_i
+      if input.is_a?(Integer) && input.positive?
+        number = input
+        break
+      else
+        puts 'Please, enter a valid input!'
+      end
+    end
+    number
+  end
+
+  def check_options(msg, options)
+    number = 0
+    loop do
+      print msg
+      input = gets.chomp.to_i
+      if options.include?(input)
+        number = input
+        break
+      else
+        puts 'Please, enter a valid input!'
+      end
+    end
+    number
+  end
+
+  def check_permission(permission)
+    case permission
+    when 'y' then permission = true
+    when 'n' then permission = false
+    end
+    permission
+  end
+
+  def create_person
+    num = check_options('Do you want to create a student (1) or a teacher (2)? [input the number]: ', [1, 2])
+    age = check_number('Age:')
+
+    print 'name:'
+    name = gets.chomp
+    classroom = check_number('Classroom:')
+
+    case num
+    when 1
+      print 'Has parent permission? [y/n]:'
+      permission = gets.chomp
+
+      permission = check_permission(permission)
+
+      @person.push(Student.new(age, name, classroom, parent_permission: permission))
+    when 2
+      print 'Specialization:'
+      specialty = gets.chomp
+
+      @person.push(Teacher.new(age, specialty, name: name))
+    else
+      puts 'Invalid number, please enter number again!'
+    end
+    puts 'Person created successfully'
+  end
+
+  def create_rental
+    puts 'Select a book from the following list'
+    @books.each_with_index do |book, index|
+      puts "#{index + 1}) Title: #{book.title}, Author: #{book.author}"
+    end
+    book_num = gets.chomp.to_i
+
+    puts 'Select a person from the following list'
+    @person.each_with_index do |per, index|
+      puts "No: #{index + 1}, [#{per.class}] Name: #{per.name}, ID: #{per.id}, Age: #{per.age}"
+    end
+    puts @person[0].name
+    iam = gets.chomp.to_i
+    puts iam
+    puts "HERE #{@person[iam - 1].name}"
+    puts @person[iam - 1].name
+    print 'Date:'
+    date = gets.chomp
+
+    p_index = iam - 1
+    p @person[p_index]
+    @rentals.push(Rental.new(date, @books[book_num - 1], @person[p_index]))
+    puts 'Rental Created successfully'
+  end
+
+  def list_rentals
+    print 'Kindly enter the ID of the person:'
+    id = gets.chomp
+    id = id.to_i
+
+    puts 'Rentals'
+
+    @rentals.each do |rental|
+      puts rental.person
+      puts "Date: #{rental.date} Book: #{rental.book.title} by #{rental.book.author}" if rental.person.id == id
+    end
```

Comment:
The rentals are suppose to be listed by date, title, and author, but your app shows the whole rental object in addition to these information as indicated below :point_down: 
Kindly check and ensure that the rental object is not displayed.

### Your UI

![Screenshot from 2023-02-02 11-12-05](https://user-images.githubusercontent.com/65233860/216311164-82988aa1-ca95-4e51-959e-999718123bd7.png)


### Actual UI

![Screenshot from 2023-02-02 11-12-40](https://user-images.githubusercontent.com/65233860/216311183-2be845d5-3114-4eb9-88c1-0ee86cfd9637.png)

Reply:
resolved! ✔️